lib/sudo-exec/src/event.rs

```diff
@@ -32,6 +51,27 @@ impl<T> EventHandler<T> {
         self.callbacks.push(callback);
     }
 
+    /// Set the handler for `SIGNAL` to be polled for read events and set `callback` to be called
+    /// if the handler is ready.  
+    pub(crate) fn set_signal_callback<const SIGNAL: c_int>(&mut self, callback: Callback<T>) {
+        let id = EventId(self.callbacks.len());
+        let Some(handler) = self.signal_handlers.get(&SIGNAL) else { 
```

Comment:
I think this should panic when SIGNAL is not found as it feels like a programmer error: "tried to use a Signal that's not in the SIGNALS set"

Reply:
I was thinking about creating a trait  `Handles<const SIGNAL: c_int> {}` that's only implemented for `EventHandler` if `SIGNAL` is in `SIGNALS`. In that way we wouldn't even need to return an `Option`.